src/scheduler_multi_app.cpp

```diff
@@ -9,6 +9,39 @@
 #include "poncos/job.hpp"
 #include "poncos/poncos.hpp"
 
+double multi_app_sched::membw_util_of_node(const size_t &idx) const {
+	double total_membw_util = 0;
+	for (size_t slot = 0; slot < SLOTS; ++slot) {
+		total_membw_util += membw_util[idx][slot];
+	}
+
+	return total_membw_util;
+}
+
+std::vector<size_t> multi_app_sched::sort_machines_by_membw_util(const std::vector<size_t> &machine_idxs,
+																 const bool reverse) const {
+	assert(machine_idxs.size() <= membw_util.size());
+
+	// determine membw_util per node
+	std::vector<double> total_membw_util;
```

Comment:
Would be more efficient if you resever memory before you use emplace_back / push_back. 

Feel free to ignore, various parts of the code are not efficient anyway. ;)